pytorch_lightning/loops/fit_loop.py

```diff
@@ -274,3 +282,11 @@ def _check_checkpoint_callback(self, should_update: bool, is_last: bool = False)
 
             for cb in callbacks:
                 cb.on_validation_end(self.trainer, model)
+
+    def state_dict(self) -> Dict:
+        if not self.is_connected:
+            raise MisconfigurationException("The Trainer should be connected to loop to retrieve the state_dict.")
+        return {self.epoch_loop.name: self.epoch_loop.state_dict()}
```

Comment:
why is the name of the attribute in this instance be determined by the class of the TrainingEpochLoop?
we can name self.my_loop anything we want. This way of state dict loading will not work for loop customization unfortunately. 

I suggest this:


`return {"epoch_loop": self.epoch_loop.state_dict()}`

This way, the class who is dumping it's state dict is responsible for knowing what it's attribute name are.

Reply:
Yes, sounds better.